### PR TITLE
Support custom plugins in the simulator

### LINKED ISSUE #8: Make it easy for users to add their custom plugins
Currently, the simulator only supports default plugins. 
But some users may want to try their custom plugins on the simulator.

- Describe in the documentation how to add a custom plugin to the simulator.
- Add some functions(or some option?) to register the custom plugins

/assign

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Support custom plugins in the simulator 🎉 
Please see `how-to-use-custom-plugins` doc in this PR.

#### Which issue(s) this PR fixes:

Fixes #8 

#### Special notes for your reviewer:

/label tide/merge-method-squash

### CODE REVIEW

File: scheduler/config/config.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package config
+
+import (
+	"k8s.io/kube-scheduler/config/v1beta2"
+	"k8s.io/kubernetes/pkg/scheduler/apis/config/scheme"
+)
+
+// DefaultSchedulerConfig creates KubeSchedulerConfiguration default configuration.
+func DefaultSchedulerConfig() (*v1beta2.KubeSchedulerConfiguration, error) {
```
Comment (sanposhiho) [inline scheduler/config/config.go:9]: Just moving from `scheduler/defaultconfig/defaultconfig.go

File: docs/how-to-use-custom-plugins/README.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+Next, add nodenumber to `AdditionalScorePlugins` since nodenumber plugin is Score plugin.
+
+```go
+func AdditionalScorePlugins() []v1beta2.Plugin {
+	var weight1 int32 = 1
+	return []v1beta2.Plugin{
+		// Note: add your original score plugins here.
+		{
+			Name:   nodenumber.Name,
+			Weight: &weight1,
```
Comment (196Ikuchil) [inline docs/how-to-use-custom-plugins/README.md:55]: What is the relationship to the weight value of KubeSchedulerConfiguration? Maybe it is good to add an explanation to the README.  (if not relevant, please ignore)

Comment (196Ikuchil) [inline docs/how-to-use-custom-plugins/README.md:55]: What is the relationship to the weight value of KubeSchedulerConfiguration? Maybe it is good to add an explanation to the README.  (if not relevant, please ignore)

File: scheduler/config/plugin.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+func DefaultFilterPluginSet() (v1beta2.PluginSet, error) {
+	defaultConfig, err := DefaultSchedulerConfig()
+	if err != nil || len(defaultConfig.Profiles) != 1 {
+		// default Config should only have default-scheduler configuration.
+		return v1beta2.PluginSet{}, xerrors.Errorf("get default scheduler configuration: %w", err)
+	}
+	return defaultConfig.Profiles[0].Plugins.Filter, nil
+}
+
+func AdditionalFilterPlugins() []v1beta2.Plugin {
```
Comment (196Ikuchil) [inline scheduler/config/plugin.go:30]: You call it `custom` in other places. Should we unify the notation?

```
func mergePluginSet(defaultPluginSet, customPluginSet v1beta2.PluginSet) v1beta2.PluginSet {
```

Comment (196Ikuchil) [inline scheduler/config/plugin.go:30]: You call it `custom` in other places. Should we unify the notation?

```
func mergePluginSet(defaultPluginSet, customPluginSet v1beta2.PluginSet) v1beta2.PluginSet {
```

**Review by 196Ikuchil — COMMENTED** [review]
Thank you! It looks nice feature.
I left some comments.
(But I'm not good at English. So, please ignore some comments if it is not needed.)

File: docs/how-to-use-custom-plugins/README.md (hunk 2 above)
Comment (sanposhiho) [reply docs/how-to-use-custom-plugins/README.md:55]: Good question. I think we can just remove this.

File: scheduler/config/plugin.go (hunk 3 above)
Comment (sanposhiho) [reply scheduler/config/plugin.go:30]: Good catch. I think it's better to call them `outOfTreePlugins`

**Comment by sanposhiho** [comment]
@196Ikuchil 
Thanks. Addressed all your reviews.

**Comment by 196Ikuchil** [comment]
Thanks! 
/lgtm

**Pull request merged**